src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -202,6 +202,11 @@ public DurableTaskExtension()
             : this(options, loggerFactory, nameResolver, orchestrationServiceFactories, shutdownNotification, durableHttpMessageHandlerFactory)
         {
             this.connectionStringResolver = connectionStringResolver;
+            if (platformInformationService == null)

```

Comment:
Why don't we move this code (205-210) into the main constructor, and then eliminate the new code you added in AzureStorageDurabilityProviderFactory?

Reply:
Not exactly sure what you mean by "main" constructor. Do you mean the DurableTaskExtension constructor in line 115? Why would that remove the need for the check in AzureStorageDurabilityProviderFactory? As I understand it, the durability provider in that constructor is still generated via DI because of this line: https://github.com/Azure/azure-functions-durable-extension/blob/95e677686000ce036f5f7c4d040f2702c8e73990/src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs#L153

So I'm not convinced that making any changes in this file would prevent the durability provider from error'ing out